ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -285,16 +290,29 @@ gcp_flush_finish([[maybe_unused]] WT_STORAGE_SOURCE *storage, [[maybe_unused]] W
     return 0;
 }
 
+// Check if the object exists in the GCP storage source.
 static int
-gcp_file_exists(
-  WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, bool *file_exists)
+gcp_file_system_exists(
+  WT_FILE_SYSTEM *file_system, [[maybe_unused]] WT_SESSION *session, const char *name, bool *existp)
 {
-    WT_UNUSED(file_system);
-    WT_UNUSED(session);
-    WT_UNUSED(name);
-    WT_UNUSED(file_exists);
+    gcp_file_system *fs = reinterpret_cast<gcp_file_system *>(file_system);
+    size_t size;
+    int ret = 0;
+    std::cout << "gcp_file_system_exists: Checking object: " << name << " exists in GCP."
+              << std::endl;
+
+    // Check whether the object exists in the cloud.
+    WT_ERR(fs->gcp_conn->object_exists(name, *existp, size));
+    if (!*existp) {
+        std::cout << "gcp_file_system_exists: Object: " << name << " does not exist in GCP."
+                  << std::endl;
+    } else
+        std::cout << "gcp_file_system_exists: Object: " << name << " exists in GCP." << std::endl;
+    return ret;
 
-    return 0;
+err:
```

Comment:
Do we need this err label? It does not seem that from til the end of the functon is used as all.

Reply:
Needed by `WT_ERR`.

Reply:
I'm not fond of using goto labels within the cpp realm, and other functions in this file seems to be directly checking the return value manually e.g. 
```
    // Get file size if the object exists.
    if ((ret = fs->gcp_conn->object_exists(gcp_fh->name, exists, size)) != 0) {
        std::cerr << "gcp_file_size: object_exists request to google cloud failed." << std::endl;
        return ret;
    }
```

Reply:
@jiechenbo I couldn't say not having authored the code. I can take a look at refactoring it.

Reply:
@marcbutler - let's refactor it, thank you!

Reply:
Refactored in [56e7f92](https://github.com/wiredtiger/wiredtiger/pull/8794/commits/56e7f92517c7fcc8bf187083a958483d9ec2fcdd)